rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/function/FunctionGenerator.xtend

```diff
@@ -448,34 +451,64 @@ class FunctionGenerator {
 				.collapseToSingleExpression(scope)
 				.mapExpression[
 					var expr = op.assignTarget(function, outs, scope)
-					for (seg : op.pathTail.indexed) {
+
+					// path intermediary
+					val intermediarySegmentSize =  op.pathTail.length - 1
+					for (var pathIndex=0; pathIndex < intermediarySegmentSize; pathIndex++) {
+						
+						val seg = op.pathTail.get(pathIndex)
+						val prop = generatePojoProperty(seg, expr.expressionType)
+						
 						val oldExpr = expr
-						val prop = (expr.expressionType as JavaPojoInterface).findProperty(seg.value.name)
 						val itemType = prop.type.itemType
-						if (seg.key < op.pathTail.size - 1) {
-							expr = JavaExpression.from(
-								'''
-								«oldExpr»
-									.«prop.getOrCreateName»(«IF prop.type.isList»0«ENDIF»)''',
-								itemType
-							)
-							if (itemType instanceof RJavaWithMetaValue) {
-								val metaExpr = expr
-								expr = JavaExpression.from('''«metaExpr».getOrCreateValue()''', itemType.valueType)
-							}
-						} else {
-							expr = JavaExpression.from(
-								'''
-								«oldExpr»
-									.«IF op.ROperationType == ROperationType.ADD»add«ELSE»set«ENDIF»«prop.name.toFirstUpper»«IF itemType instanceof RJavaWithMetaValue && !op.assignAsKey»Value«ENDIF»(«it»)''',
-								JavaPrimitiveType.VOID
-							)
+						expr = JavaExpression.from(
+							'''
+							«oldExpr»
+								.«prop.getOrCreateName»(«IF prop.type.isList»0«ENDIF»)''',
+							itemType
+						)
+						if (itemType instanceof RJavaWithMetaValue) {
+							val metaExpr = expr
+							expr = JavaExpression.from('''«metaExpr».getOrCreateValue()''', itemType.valueType)
 						}
 					}
+					
+					//end of path
+					val seg = op.pathTail.get(op.pathTail.length - 1)
+					val prop = generatePojoProperty(seg, expr.expressionType)
+					val oldExpr = expr
+					val itemType = prop.type.itemType
+					if (expr.expressionType.itemType instanceof RJavaWithMetaValue) {
+						expr = JavaExpression.from(
+							'''
+							«oldExpr»
+								.«prop.getOrCreateName»().set«seg.name.toFirstUpper»(«it»)''',
```

Comment:
Not sure if it matters: instead of `seg.name.toFirstUpper`, `prop.name.toFirstUpper`? That's consistent with the code below.